poshc2/server/Payloads.py

```diff
@@ -238,112 +167,107 @@ def CreateDroppers(self, name=""):
+            patch = payload.decode("utf-8")
+            patchlen = 8000 - len(patch)
 
-        # Load CLR "v4.0.30319"
-        self.QuickstartLog("" + Colours.END)
-        self.QuickstartLog("ReflectiveDLL that loads C# Implant in CLR v4.0.30319 - DLL Export (VoidFunc)" + Colours.GREEN)
-        with open('%sSharp_v4_x86_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v4_86 = f.read()
-        self.PatchSharpBytes("%sSharp_v4_x86.dll" % name, v4_86, offsetFinder('%sSharp_v4_x86_dll.b64' % PayloadTemplatesDirectory), "")
-        with open('%sSharp_v4_x64_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v4_64 = f.read()
-        self.PatchSharpBytes("%sSharp_v4_x64.dll" % name, v4_64, offsetFinder('%sSharp_v4_x64_dll.b64' % PayloadTemplatesDirectory), "")
+        patch2 = ""
+        patch2 = patch2.ljust(patchlen, '\x00')
+        patch3 = "%s%s" % (patch, patch2)
 
-        # Load CLR "v4.0.30319"
-        self.QuickstartLog("" + Colours.END)
-        self.QuickstartLog("ReflectiveDLL that loads PBind C# Implant in CLR v4.0.30319 - DLL Export (VoidFunc)" + Colours.GREEN)
-        with open('%sPosh_v4_x86_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v4_86 = f.read()
-        self.PatchPBindBytes("%sPBind_v4_x86.dll" % name, v4_86, offsetFinder('%sPosh_v4_x86_dll.b64' % PayloadTemplatesDirectory), "")
-        with open('%sPosh_v4_x64_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v4_64 = f.read()
-        self.PatchPBindBytes("%sPBind_v4_x64.dll" % name, v4_64, offsetFinder('%sPosh_v4_x64_dll.b64' % PayloadTemplatesDirectory), "")
+        f = open(filename, "r+b")
+        f.seek(offset)
+        f.write(bytes(patch3, 'UTF-16LE'))
+        f.close()
 
-        # End of logging for dll
-        self.QuickstartLog(Colours.END)
-        self.QuickstartLog("RunDLL Example:" + Colours.GREEN)
-        self.QuickstartLog("rundll32 Sharp_v4_x64.dll,VoidFunc")
+        self.QuickstartLog("Payload written to: %s" % (filename))
 
-    def CreateShellcode(self, name=""):
-        # Load CLR "v2.0.50727"
-        self.QuickstartLog(Colours.END)
-        self.QuickstartLog("Shellcode that loads CLR v2.0.50727" + Colours.GREEN)
-        with open('%sPosh_v2_x86_Shellcode.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v2_86 = f.read()
-        self.PatchBytes("%sPosh_v2_x86_Shellcode.bin" % name, v2_86, offsetFinder('%sPosh_v2_x86_Shellcode.b64' % PayloadTemplatesDirectory), "Shellcode")
-        with open("%s%sPosh_v2_x86_Shellcode.bin" % (self.BaseDirectory, name), 'rb') as binary:
-            with open("%s%sPosh_v2_x86_Shellcode.b64" % (self.BaseDirectory, name), 'wb') as b64:
+    def CreateDll(self, DestinationFile, ResourceFile, DllName):                
+        with open(ResourceFile, 'r') as f:
+            fileRead = f.read()
+        self.PatchBytes(DestinationFile, fileRead, offsetFinder(ResourceFile), "DLL")
```

Comment:
Should we be passing the `DllName` variable here? Otherwise it's unused.

Reply:
Yeah that was a bit bizzare, fixed that